components/CreateAPIKeys.tsx

```diff
@@ -14,48 +13,71 @@ import {
 } from '@/components/ui/dialog';
 import { Input } from '@/components/ui/input';
 import { Label } from '@/components/ui/label';
-import { createClient } from '@/utils/supabase/client';
 import { useAuthStore } from '@/store/useAuthStore';
-import { useOrgsStore } from '@/store/useOrgsStore';
 import { useToast } from '@/components/ui/use-toast';
 import Loader from '@/components/ui/Loader';
+import { OrganizationMember } from '@/hooks/useOrganizationMember';
+import { requestCarbon } from '@/utils/carbon';
 
-const supabase = createClient();
+// TODO move this interface into a dedicated file
+export interface APIKey {
+    id: number;
+    token_hash: string;
+    description?: string;
+    customer_id: number;
+    customer_email: string;
+    organization_id: number;
+    organization_name: string,
+    expires_at: Date,
+    created_at: Date,
+    updated_at: Date,
+}
 
-function CreateAPIKeys() {
+const CreateAPIKeys = (
+    props: {
+        organizationMember: OrganizationMember,
+        encryptedId: string,
+    },
+) => {
     const [label, setLabel] = useState('My New Key');
     const [key, setKey] = useState('');
     const [isLoading, setIsLoading] = useState(false);
+    const [isDialogOpen, setIsDialogOpen] = useState(false);
+
     const { user } = useAuthStore();
     const { toast } = useToast();
-    const { activeOrg } = useOrgsStore();
-    const [isDialogOpen, setIsDialogOpen] = useState(false);
 
     const createAPIKey = async () => {
-        if (!label || !user || !activeOrg) return;
-        setIsLoading(true); // Start loading
-        const newKey = uuidv4();
-        const { error } = await supabase.from('api_keys').insert([
-            {
-                key: newKey,
-                label,
-                created_by: user.id,
-                org_id: activeOrg.org_id,
+        if (!label || !user ) {
+            return;
```

Comment:
I am not sure exactly how this works but an empty return can return in weird experience where you click on something and nothing happens
Better to show a toast/error and then return
Unless there are already validations and this is just a fallback